flat-manager-client

```diff
@@ -992,6 +993,17 @@ async def run_with_session(args):
     return result
 
 
+def check_token(token):
+    """
+    Fails if token is expired
+    """
+    base64_data = token.split(".")[1]
+    payload = json.loads(base64.urlsafe_b64decode(base64_data).decode("utf-8"))
+    expiration = datetime.datetime.fromtimestamp(payload["exp"], datetime.timezone.utc)
+    if datetime.datetime.now(datetime.timezone.utc) > expiration:
+        raise SystemExit(f"Token {payload} has expired")
```

Comment:
probably better to to just print out the expiry date here than the JSON blob

Reply:
The intent here is JSON blob contains information what exactly should be requested. It may have one or multiple prefixes, one or multiple branches etc.

Reply:
It also contains the name of token that was created.

Reply:
yes, but why would need that information in a "token has expired" error message? the error is that it expired.

if you want to log the capabilities of the token it should be somewhere else, not in the error message.

Reply:
Ok.

Reply:
I'd just log them whenever token is used: the ref prefix, target repo and capabilities. Probably not as json though, as that makes it harder to read inside ci logs.

Reply:
Sounds like fairly invasive changes. If someone else wants to pick that up, fine. I think touching non-standard fields in blobs in client is actively dangerous since client might not be at all same version as server. JWT is for most intents and purposes supposed to be considered opaque blob by clients. Exp just happens to be a standard field and useful client-side.

Reply:
I went ahead and improved the server's response a bit in case of expiry https://github.com/flatpak/flat-manager/pull/144

![image](https://github.com/user-attachments/assets/4b73348b-b0f4-4e96-9c54-9c7e1576679f)